libraries/api-extractor-model/src/mixins/ApiItemContainerMixin.ts

```diff
@@ -209,6 +274,173 @@ export function ApiItemContainerMixin<TBaseClass extends IApiItemConstructor>(
+              membersToAdd.push(member);
+            }
+          }
+        }
+
+        for (const member of membersToAdd) {
+          if (ApiNameMixin.isBaseClassOf(member)) {
+            const members: ApiItem[] = membersByName.get(member.name) || [];
+            members.push(member);
+            membersByName.set(member.name, members);
+          } else {
+            const members: ApiItem[] = membersByKind.get(member.kind) || [];
+            members.push(member);
+            membersByKind.set(member.kind, members);
+          }
+        }
+
+        // Interfaces can extend multiple interfaces, so iterate through all of them.
+        const extendedItems: ApiItem[] = [];
+        let extendsTypes: readonly HeritageType[] | undefined;
+
+        switch (next.kind) {
+          case ApiItemKind.Class: {
+            const apiClass: ApiClass = next as ApiClass;
+            extendsTypes = apiClass.extendsType ? [apiClass.extendsType] : [];
+            break;
+          }
+          case ApiItemKind.Interface: {
+            const apiInterface: ApiInterface = next as ApiInterface;
+            extendsTypes = apiInterface.extendsTypes;
+            break;
+          }
+        }
+
+        if (extendsTypes === undefined) {
+          messages.push({
+            messageId: FindApiItemsMessageId.UnsupportedKind,
+            text: `Item ${next.displayName} is of unsupported kind ${next.kind}.`
+          });
+          maybeIncompleteResult = true;
+          break;
+        }
+
+        for (const extendsType of extendsTypes) {
+          // We want to find the reference token associated with the actual inherited declaration.
+          // In every case we support, this is the first reference token. For example:
+          //
+          // ```
+          // export class A extends B {}
+          //                        ^
+          // export class A extends B<C> {}
+          //                        ^
+          // export class A extends B.C {}
+          //                        ^^^
+          // ```
+          const firstReferenceToken: ExcerptToken | undefined = extendsType.excerpt.spannedTokens.find(
+            (token: ExcerptToken) => {
+              return token.kind === ExcerptTokenKind.Reference && token.canonicalReference;
+            }
+          );
+
+          if (!firstReferenceToken) {
+            messages.push({
+              messageId: FindApiItemsMessageId.UnexpectedExcerptTokens,
+              text: `Encountered unexpected excerpt tokens in ${next.displayName}. Excerpt: ${extendsType.excerpt.text}.`
+            });
+            maybeIncompleteResult = true;
+            continue;
+          }
+
+          const apiModel: ApiModel | undefined = this.getAssociatedModel();
+          if (!apiModel) {
+            messages.push({
+              messageId: FindApiItemsMessageId.MissingApiModel,
+              text: `Unable to get the associated model of ${next.displayName}.`
```

Comment:
This might arise if the `ApiItem` object was newly constructed but not added to an `ApiModel` yet. The problem isn't that we can't get the model, it's that no model has been associated.  Maybe we could word it like:

"Unable to analyze references of API item ${next.displayName} because it is not associated with an ApiModel."

Reply:
Took this verbatim, see https://github.com/microsoft/rushstack/pull/3545/files#diff-f0516a1941ffe40f35eb325cb731d6d3af1663276d61824a4a6da636d91d278eR392-R393.